### ISSUE TITLE
Dark themes not applying to left and top bar

### ISSUE THREAD

**Issue by jensuffhaus** [danielperna84/hass-configurator#199]
Selecting a dark theme in the settings for the editor only applies to the actual text edit panel and not the side or top panels of the editor.

![image](https://user-images.githubusercontent.com/1774160/83352406-66281900-a34b-11ea-9583-b58fcc82a5aa.png)

![image](https://user-images.githubusercontent.com/1774160/83352425-940d5d80-a34b-11ea-9930-f329e057e114.png)

**Comment by danielperna84** [comment]
Yes, the scope of the available themes is limited to the editor. The whole configurator app itself does not support theming, and there are no plans to add support from my side. I'd be open to contributions by others though.

**Comment by killtux** [comment]
@danielperna84 is it possible to realize a simple dark mode that automatically switches like Home Assistant interface itself or switch if its a bright/dark theme? that white really burns into your eyes if every app and site on your computer is dark 😅.

i am afraid i don't have the skills to contribute the necessary code changes, sorry for that.

**Comment by danielperna84** [comment]
@killtux 

If someone volunteers to implement it, sure. The main problem is, that the UI-framework used in this project still was in beta-state back then. Switching to a more recent version requires some effort. And even then I don't know if the framework has builtin support for a dark mode.

Anyways, I won't take care of this issue. But anybody is welcome to do so.

**Comment by Vince1024** [comment]
Hi,
Mine is not taking any count of the chosen theme at all event the center part of the UI.
Is it possible to get the current HA theme taking in count in File editor ?
Thanks

**Comment by danielperna84** [comment]
@Vince1024 no, as explained further above, the File Editor does not support theming.

**Comment by igorsantos07** [comment]
I'm wondering if I can spend some time trying to work this out... It's something that bothers me very much and VSCode is terrible on my RPi. Here's what I could summarize and would like some opinion :)

Materialize is sadly defunct, but they released a v1 before dying. Here's the [upgrade guide](https://github.com/Dogfalo/materialize/blob/v1-dev/v1-upgrade-guide.md), but given there's nothing new still being launched, I don't think it's worth upgrading? @danielperna84 if you could take a quick look on it and see if you can spot used UI elements that could benefit from the upgrade...

Also, Daniel, I'm thinking how this could be implemented. To my poor understanding of HA, the add-on isn't able to get which theme HA is using, and thus it would only change color based on the code theme selected. Seems right? And if so, do you think it would be feasible to sniff out if the theme is light or dark (maybe even based on a manual list) to then decide if the external UI should be light/dark? I guess, running a theme checker on startup (for existing setups) and on theme change. I'm asking this because I haven't checked almost anything from the source to see how it works under the hood.

And then, I _hope_ it's possible to apply light/dark based on CSS variables, either coming from a system-wide class or can changed programmatically in the `style` prop from `<html>`. I say "hope" because it depends on how colors are setup via Materialize... but given our palette is quite simple an the whole project is pretty much frozen, hardcoded new colors wouldn't hurt?

**Comment by danielperna84** [comment]
@igorsantos07 

I haven't followed what has happened to materialize. But I think it should be considered to switch to a different UI framework. A lot of websites select their mode based on the OS settings. Just like HA does. So it would make sense to see how HA detects the OS-setting, and within the configurator provide a select-box with the values "bright", "dark" and "os".

That being said, this would only apply to the blue and white areas around the editor. The editor has its own theming. And I don't know how that could be kept in sync with a potential dynamic style when in OS-mode.

**Comment by igorsantos07** [comment]
I remember seeing JS APIs and media-queries to automatically identify the default dark/light mode - never used them but I know they exist, it's even possible to emulate that from DevTools. I'll try going that route, then. Having the option to manually select it is important anyway, as for instance my OS doesn't properly report dark mode :joy: 

And yep, I agree that moving out of Materialize is the way to go in the long run, but definitely out of the scope for such a "simple" problem as this issue. That's great once some time is found to rework some of the interface and upgrade other components of it :)

And thanks for the quick answer!

**Comment by danielperna84** [comment]
@igorsantos07 

I just found this [community-maintained fork of Materialize](https://github.com/materializecss/materialize), which also has theme support: https://materializeweb.com/themes.html

Maybe that's an option?

**Comment by mallewski** [comment]
I've opened a pull request that fixes this: #253

It adds a full dark mode for the entire UI (navbar, sidebar, filepath bar,
panels, dropdowns) with auto/light/dark toggle and server-side persistence
so the setting works across all devices.

A manual workaround (without waiting for the merge) is described in the PR.
